third_party/txt/src/txt/paragraph.cc

```diff
@@ -180,31 +177,78 @@ void Paragraph::SetText(std::vector<uint16_t> text, StyledRuns runs) {
   runs_ = std::move(runs);
 }
 
-void Paragraph::InitBreaker() {
-  breaker_.setLocale(icu::Locale(), nullptr);
-  breaker_.resize(text_.size());
-  memcpy(breaker_.buffer(), text_.data(), text_.size() * sizeof(text_[0]));
-  breaker_.setText();
-}
-
-bool Paragraph::AddRunsToLineBreaker(
-    std::unordered_map<std::string, std::shared_ptr<minikin::FontCollection>>&
-        collection_map /* TODO: Cache the font collection here. */) {
-  minikin::FontStyle font;
-  minikin::MinikinPaint paint;
-  for (size_t i = 0; i < runs_.size(); ++i) {
-    auto run = runs_.GetRun(i);
-    GetFontAndMinikinPaint(run.style, &font, &paint);
-    auto collection = font_collection_->GetMinikinFontCollectionForFamily(
-        run.style.font_family);
-    if (collection == nullptr) {
-      FXL_LOG(INFO) << "Could not find font collection for family \""
-                    << run.style.font_family << "\".";
-      return false;
+bool Paragraph::ComputeLineBreaks() {
+  line_ranges_.clear();
+  line_widths_.clear();
+
+  std::vector<size_t> newline_positions;
+  for (size_t i = 0; i < text_.size(); ++i) {
+    if (text_[i] == '\n')
```

Comment:
You need to handle everything in the newline character set. Let me try to fetch the ICU method that lets you do this.

Reply:
Changed this to query ICU for characters with the linefeed or mandatory break properties